### DISCUSSION TITLE
acsess to private project in row format in url web via token

### DISCUSSION THREAD

**Discussion by Hamed-Republic** [community/community#55801] [Code Security] [answered]
### Select Topic Area

Question

### Body

hi guys how can acsess private project row format  via token in web url ?
please help me 
thanks

**Comment by airtower-luna** [comment]
You don't. The tokens in URL you see when you access the raw file via web interface are temporary ones.

If you want to use a PAT or other OAuth token, you need to send the token in the `Authorization` header, as described here: https://docs.github.com/en/rest/overview/authenticating-to-the-rest-api?apiVersion=2022-11-28

**Comment by Hamed-Republic** [comment]
That is, there is no way؟
This is sad.I need to access safe my private project via  link as raw format

**Comment by airtower-luna** [comment]
Nothing *persistent*. The links with tokens you get when viewing the raw file in the browser will work only for a short time. If you want persistent access, you need the same URL *minus token* and send a valid token in the `Authorization` header.

**Comment by Hamed-Republic** [comment]
Could you write me an example url via Authorization ?

**Comment by airtower-luna** [comment]
The URL is exactly the same, the `https://raw.githubusercontent.com/OWNER/REPO/REF/PATH` kind. You need to set the `Authorization` *header* in the request to `Bearer YOUR_TOKEN_HERE`. How to do that depends on the HTTP client/library you use.

**Reply by Hamed-Republic** [reply]
i tested this  link and and not worked 
https://raw.githubusercontent.com/Hamed-Republic/node2/main/README.md?Authorization=ghp_F6jV1J9h4YkrlynGFH******G1mitKjx1viHWF
I am not familiar with the HTTP silent library
Is the combination correct?

**Reply by airtower-luna** [reply, marked as answer]
No, that's not correct. You *can't put the token into the URL*. If you want to send someone a link, or use `<a href="...">`, that won't work.

The HTTP client is the thing you use to make the HTTP request (to download the file). It could be something like `curl`, a programming library (e.g. `requests` in Python), or a browser.
